src/bin/casr-cluster.rs

```diff
@@ -408,10 +408,77 @@ fn dedup(indir: &Path, outdir: Option<PathBuf>) -> Result<(usize, usize)> {
     Ok((before, after))
 }
 
+/// Merge sub directory to main
+///
+/// # Arguments
+///
+/// * `new` - path to directory with new CASR reports
+///
+/// * `storage` - path to storage directory with CASR reports
+///
+/// # Return value
+///
+/// Number of merged reports
+fn merge_dirs(new: &Path, storage: &Path) -> Result<u64> {
+    let dir = fs::read_dir(storage).with_context(|| {
+        format!(
+            "Error occurred while opening directory with Casr reports. File: {}",
+            storage.display()
+        )
+    })?;
+
+    let mut mainhash = HashSet::new();
+    for entry in dir.flatten() {
+        if entry.path().extension().is_some() && entry.path().extension().unwrap() == "casrep" {
+            if let Ok(trace) = stacktrace(entry.path().as_path()) {
+                mainhash.insert(trace);
+            } else {
+                bail!("Storage directory corrupted, merge failed.");
+            }
+        }
+    }
+
+    let dir = fs::read_dir(new).with_context(|| {
+        format!(
+            "Error occurred while opening directory with Casr reports. File: {}",
+            new.display()
+        )
+    })?;
+
+    let mut new: u64 = 0;
+    for entry in dir.flatten() {
+        if entry.path().extension().is_some() && entry.path().extension().unwrap() == "casrep" {
+            if let Ok(trace) = stacktrace(entry.path().as_path()) {
+                if mainhash.insert(trace) {
+                    let target = Path::new(&storage).join(entry.file_name());
+                    if target.exists() {
+                        println!(
+                            "File with name {} already exists in STORAGE_DIR.",
```

Comment:
Maybe, we can generate a new name, like web browser does name-1, name-2...

Reply:
Maybe, better user should decide? Contents of the report often associated with name. I think, it is an acceptable solution for now.

Reply:
agree with @anfedotoff